src/javascript/app/pages/cashier/payment_agent_list.js

```diff
@@ -94,8 +94,16 @@ const PaymentAgentList = (() => {
     };
 
     const showEmptyListMsg = () => {
-        $('.barspinner').setVisibility(0);
-        $('#no_paymentagent').setVisibility(1);
+        const currency = Client.get('landing_company_shortcode');
+        if (currency === 'virtual') {
+            $('.barspinner').setVisibility(0);
+            $('#virtual_currency').setVisibility(1);
+            $('#no_paymentagent').setVisibility(0);
+        } else {
+            $('.barspinner').setVisibility(0);
+            $('#virtual_currency').setVisibility(0);
+            $('#no_paymentagent').setVisibility(1);
```

Comment:
Please use our getElementById function to select a dom by id.
```const getElementById     = require('./common_functions').getElementById;```

Please check if we can handle this using `data-show`

Reply:
I could indeed make use of data-show handles. It should be fixed now.